src/hooks/useToggle.js

```diff
@@ -0,0 +1,17 @@
+import {useState} from 'react';
+
+/**
+ * Custom hook for state variables that have a special boolean state only
+ * @param {*} initialValue - initial boolean value of the state variable
+ */
+const useToggle = initialValue => {
+    const [state, setState] = useState(initialValue);
+
+    const toggleState = () => {
+        setState(!state);
```

Comment:
I think it will work all the same, but would it not be better to use [functional updates](https://reactjs.org/docs/hooks-reference.html#functional-updates)?

Reply:
I don't see how that would work here, can you post the code for this case?

Reply:
```suggestion
        setState(state => !state);
```

Reply:
What is the advantage of that / Disadvantage of the existing solution?

Reply:
Also, kinda feel like we should use `useCallback` here

Reply:
Or something similar, to avoid creating the `toggleState` function repeatedly, which might trigger re-renders

Reply:
Its only dependency is `setState`, which is guaranteed (as per the [hooks api docs - see note for `useState`](https://reactjs.org/docs/hooks-reference.html#usestate)) to not change.

Reply:
Now that's a good thing to have learned! Btw, the hook does not need dependencies in this case, the callback does not change, because it depends strictly from the argument, `state` in this case, so it looks like this:
```js
const toggleState = useCallback(() => {
        setState((state) => !state);
    }, []);
```

By doing this, not only we optimize performace, but we also fix a problem, because currently, if there is more than 1 variable created with `useToggle`, there would be infinite re-renders, which is solved with this `useCallback` solution.

kudos for this catch!

Reply:
That was exactly my point, that's it!

I got that from the [`react-redux` docs on migrating to hooks](https://react-redux.js.org/next/api/hooks). They are pretty helpful when considering some of the caveats of hooks, because of functions being recreated since they are being declared in `render()`.
It's a good read  for both the React and the Redux side of things, highly recommend it 🙂